domain/keyupdater/service/service.go

```diff
@@ -113,15 +123,35 @@ func (s *Service) GetAuthorisedKeysForMachine(
 ) ([]string, error) {
 	if err := machineName.Validate(); err != nil {
 		return nil, fmt.Errorf(
-			"cannot get authorised keys for machine %q: %w",
+			"validating machine name when getting authorized keys for machine: %w",
+			err,
+		)
+	}
+
+	err := s.st.CheckMachineExists(ctx, machineName)
+	if errors.Is(err, machineerrors.MachineNotFound) {
```

Comment:
Not sure if we need to trace this error. The block in `else if` does everything already.

Reply:
We are re-writing the error to the new context and taking the information we don't need as it is well understood and defined. See the errors email for more info.